src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -246,6 +247,8 @@ public void postInject() {
                         cloudCallThreadPool));
         greengrassCoreIPCService.setAuthorizeClientDeviceActionHandler(
                 context -> new AuthorizeClientDeviceActionOperationHandler(context, serviceApi, authorizationHandler));
+        greengrassCoreIPCService.setPutComponentMetricHandler(
```

Comment:
why are you doing this? `putComponentMetric` isn't implemented in CDA, it is implemented in the Nucleus.

Reply:
I talked to Nikkhil last week and he mentioned I should write it without the authorization portion of `putComponentMetric`, so I made the class without those bits

Reply:
This is unworkable. We need a change in the Nucleus to support our use case. Copying the implementation is not the right thing to do.